segmentstore/server/src/main/java/io/pravega/segmentstore/server/writer/SegmentAggregator.java

```diff
@@ -1093,6 +1099,37 @@ private void mergeCompleted(SegmentProperties segmentProperties, UpdateableSegme
                 }, this.executor);
     }
 
+    /**
+     * Deletes a Segment and its Attributes from Storage. This can be used to delete either the Segment handled by this
```

Comment:
I'm confused by the contract of this method. It says that:

> This can be used to delete either the Segment handled by this instance or any other Segment that has a pending Merge into it.

Checking the contract of `storage.delete(handle, timer.getRemaining())`, I see:

```
/**
     * Deletes a StreamSegment.
     *
     * @param handle  A read-write SegmentHandle that points to a Segment to Delete.
     * @param timeout Timeout for the operation.
     * @return A CompletableFuture that, when completed, will indicate the operation succeeded. If the operation failed,
     * it will contain the cause of the failure. Notable exceptions:
     * <ul>
     * <li> StreamSegmentNotExistsException: When the given Segment does not exist in Storage.
     * <li> StorageNotPrimaryException: When this Storage instance is no longer primary for this Segment (it was fenced out).
     * </ul>
     * @throws IllegalArgumentException If handle is read-only.
     */
    CompletableFuture<Void> delete(SegmentHandle handle, Duration timeout);

```

which does not refer to other segments that have a pending merge. I also see that `dataSource.deleteAllAttributes(metadata, timer.getRemaining()))` has the following contract:

```
/**
     * Instructs the DataSource to delete the Attribute Index for the given Segment.
     *
     * @param segmentMetadata The metadata for the Segment to delete attributes for.
     * @param timeout         Timeout for the operation.
     * @return A CompletableFuture that, when completed, will indicate that the operation completed. If the operation
     * failed, the Future will complete with the appropriate exception.
     */
    CompletableFuture<Void> deleteAllAttributes(SegmentMetadata segmentMetadata, Duration timeout);
```

which only talks about the attribute index of the segment, not merger segments. Consequently, I'm not being able to validate that this is implementing the expected behavior. Could you clarify, pls?

Reply:
I reworded the javadoc.